repository/src/main/java/uk/gov/hmcts/ccd/definition/store/repository/interceptor/CaseRoleInterceptor.java

```diff
@@ -0,0 +1,20 @@
+package uk.gov.hmcts.ccd.definition.store.repository.interceptor;
+
+import org.hibernate.EmptyInterceptor;
+import org.hibernate.type.Type;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.definition.store.repository.entity.CaseRoleEntity;
+
+import java.io.Serializable;
+
+@Component
+public class CaseRoleInterceptor extends EmptyInterceptor {
+    @Override
+    public boolean onSave(Object entity, Serializable id, Object[] state, String[] propertyNames, Type[] types) {
+        if (entity instanceof CaseRoleEntity){
+            CaseRoleEntity caseRoleEntity = (CaseRoleEntity) entity;
+            caseRoleEntity.setReference(caseRoleEntity.getReference().toUpperCase());
```

Comment:
do we need to introduce a complex element as an interceptor to do the uppercase transformation? 
Can't we have it somewhere else?

Reply:
I had a similar doubt. I thought of adding it to the setter of the entity, however that brings some artificial controls on the tests and parser classes. The interceptor seems the best place regarding the side effects though seems like a bit over use...